core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -813,6 +813,17 @@ public void testCorrelatedJoinWithTopN()
                                         anyTree(tableScan("nation", ImmutableMap.of("nation_name", "name", "nation_regionkey", "regionkey")))))))));
     }
 
+    @Test
+    public void testCorrelatedJoinWithNullCondition()
+    {
+        assertPlan(
+                "SELECT regionkey, n.name FROM region LEFT JOIN LATERAL (SELECT name FROM nation) n ON NULL",
+                any(
+                        join(LEFT, builder -> builder
```

Comment:
please add a join filter in the expected plan

Reply:
I meant the `filter`, not the `equiCriteria`. However, the filter is empty at that planning phase, as the result of PredicatePushDown. To see how the filter is planned initially, please test with `CREATED`.

Reply:
The filter is empty so cannot assert that it's there, since it isn't.
Sure, will add another case with `CREATED`